File: 2Lab/homework/homework.cpp
#include <iostream>
#include <cmath>
#include <vector>

class IntMatrix {
private:
    std::vector<std::vector<int>> data;

public:
    IntMatrix(const std::vector<std::vector<int>>& matrix) {
        data = matrix;
    }

    
    double getRowRMS(int row) const {
        if (row < 0 || row >= data.size()) {
            throw std::out_of_range("Invalid row index");
        }

        double sumOfSquares = 0.0;
        for (int value : data[row]) {
            sumOfSquares += value * value;
        }

        return std::sqrt(sumOfSquares / data[row].size());
    }

    
    int getTotalSum() const {
        int total = 0;
        for (const auto& row : data) {
            for (int value : row) {
                total += value;
            }
        }
        return total;
    }
};

int main() {
    std::vector<std::vector<int>> matrixData = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };

    IntMatrix matrix(matrixData);

    std::cout << "Root mean square of each row:\n";
    for (int i = 0; i < matrixData.size(); ++i) {
        std::cout << "Row " << i << ": " << matrix.getRowRMS(i) << "\n";
    }

    std::cout << "\nTotal sum of all elements: " << matrix.getTotalSum() << "\n";

    return 0;
}

File: 2Lab/homework/homework.cs
using System;

class IntMatrix
{
    private int[,] data;

    public IntMatrix(int[,] initialData)
    {
        data = initialData;
    }

    
    public double this[int row]
    {
        get
        {
            if (row < 0 || row >= data.GetLength(0))
                throw new IndexOutOfRangeException("Invalid row index");

            int columns = data.GetLength(1);
            double sumOfSquares = 0;

            for (int j = 0; j < columns; j++)
            {
                sumOfSquares += data[row, j] * data[row, j];
            }

            return Math.Sqrt(sumOfSquares / columns);
        }
    }

    
    public int TotalSum
    {
        get
        {
            int sum = 0;
            for (int i = 0; i < data.GetLength(0); i++)
            {
                for (int j = 0; j < data.GetLength(1); j++)
                {
                    sum += data[i, j];
                }
            }
            return sum;
        }
    }
}

class Program
{
    static void Main()
    {
        int[,] matrixData = {
            { 1, 2, 3 },
            { 4, 5, 6 },
            { 7, 8, 9 }
        };

        IntMatrix matrix = new IntMatrix(matrixData);

        Console.WriteLine("Root mean square of each row:");
        for (int i = 0; i < matrixData.GetLength(0); i++)
        {
            Console.WriteLine($"Row {i}: {matrix[i]:F2}");
        }

        Console.WriteLine($"\nTotal sum of all elements: {matrix.TotalSum}");
    }
}
